### PR TITLE
Refactor class methods to arrow functions

### PR DESCRIPTION
**Pull request by nikita-rudenko**
Happy Hacktoberfest :tada:
This PR is related to https://github.com/dmtrKovalenko/date-io/issues/84

I found all classes and turned their methods into arrow functions.

### CODE REVIEW

File: packages/date-fns-jalali/src/date-fns-jalali-utils.ts [hunk 1]
```diff
@@ -191,83 +191,83 @@ export default class DateFnsJalaliUtils implements IUtils<Date> {
...
-  }
+  };
 
-  public setYear(value: Date, count: number) {
+  public setYear = (value: Date, count: number) => {
     return setYear(value, count);
-  }
+  };
 
   public date(value?: null): null;
```
Comment (nikita-rudenko) [inline packages/date-fns-jalali/src/date-fns-jalali-utils.ts:272]: @dmtrKovalenko I am not sure how to refactor this

Comment (dmtrKovalenko) [inline packages/date-fns-jalali/src/date-fns-jalali-utils.ts:272]:
```suggestion
```

This line can be removed. Thanks :)

**Comment by dmtrKovalenko** [comment]
Giant thanks for this. One is left – check that class properties are properly transpired.

**Comment by nikita-rudenko** [comment]
Seems fine. The main difference is that class methods were transpiled as prototype methods and now with arrow functions it uses `this`.

**Comment by dmtrKovalenko** [comment]
The main idea to make destruction works like on the simple objects.

**Comment by nikita-rudenko** [comment]
Do you mean that it should be accesible via named imports?

I created a file for a test in the root:
```
import DateFNS from './packages/date-fns/build/index'

DateFNS.prototype.addDays()
```

According to autocompletion, it works the same with a build before and after my changes :thinking:

**Comment by dmtrKovalenko** [comment]
No, idea to make possible to do 
```ts
const adapter = new DateFnsAdapter()
const { addDays } = adapter
```

But anyway it is a useful thing to avoid confusion with call context!

**Comment by nikita-rudenko** [comment]
I doubt that approach for testing is the best, but anyway this is what I get with a build after my changes:

Screenshot from 2020-10-08 21-51-20

**Pull request merged**